### PR TITLE
Grpc monitor

### PR DESCRIPTION
**Pull request by autholykos**
This PR is about the migration from the clunky unixSupervisor monitoring to a grpc one based on the structs defined in dusk-networ/dusk-protobuf. 

Please note that this PR is not intended for the wholesome monitoring system, but rather just migrate the functionalities that were already implemented through the `eventmon` package. This work serves as a starting point to implement a better monitoring system in the near future

This PR depends on dusk-network/dusk-protobuf#26 to be merged.

### CODE REVIEW

**Comment by autholykos** [comment]
Failing test `TestNotifySlowdown` is caused by excessively strict timeout management

File: pkg/eventmon/grpc/client_test.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	// This function is blocking, so we run it in a goroutine
+	go grpcServer.Serve(l)
+	return hs
+}
+
+// TestMain automates testing of BlockUpdates received through the grpc call. It
+// accepts a clientMethod to prep the test, and a varargs of tester functions
+// which apply to the payload received. Each tester is supposed to test a
+// correspondent payload
+func Suite(t *testing.T, timeoutMillis time.Duration, calls ...callTest) {
```
Comment (devtomnl) [inline pkg/eventmon/grpc/client_test.go:59]: func comment for exported methods should start with method name, eg: `// Suite ... `

**Comment by autholykos** [comment]
blocked by #380

**Comment by autholykos** [comment]
fix #348

**Pull request merged**